File: scripts/05_alphafold3_analysis/00_config.R
# 00_config.R
# Central R configuration for AlphaFold3 analysis summary scripts.

suppressPackageStartupMessages({
  library(fs)
})

PROJECT_DIR <- normalizePath(getwd(), mustWork = FALSE)

METADATA_DIR <- file.path(PROJECT_DIR, "metadata")
OUT_DIR <- file.path(PROJECT_DIR, "results", "05_alphafold3_analysis")
INPUT_DIR <- file.path(OUT_DIR, "input")
AF3_OUTPUT_DIR <- file.path(INPUT_DIR, "af3_outputs")

AF3_DESIGNS <- file.path(METADATA_DIR, "af3_designs.tsv")
MOTIF_POSITION_MAP <- file.path(METADATA_DIR, "af3_motif_position_map.tsv")
BETA2_RESIDUE_MAP <- file.path(METADATA_DIR, "af3_beta2_residue_map.tsv")

MANIFEST <- file.path(OUT_DIR, "af3_output_manifest.tsv")
CONFIDENCE_SUMMARY <- file.path(OUT_DIR, "af3_confidence_summary.tsv")
DNA_RMSD <- file.path(OUT_DIR, "dna_rmsd_by_model.tsv")
CONTACTS <- file.path(OUT_DIR, "protein_dna_base_contacts.tsv")
BETA2_DISTANCES <- file.path(OUT_DIR, "beta2_to_dna_base_min_distances.tsv")

dir_create(OUT_DIR, recurse = TRUE)
dir_create(file.path(OUT_DIR, "plots"), recurse = TRUE)

message("PROJECT_DIR: ", PROJECT_DIR)
message("OUT_DIR: ", OUT_DIR)

File: scripts/05_alphafold3_analysis/00_config.py
#!/usr/bin/env python3

"""Central Python configuration for AlphaFold3 analysis scripts."""

from pathlib import Path

PROJECT_DIR = Path.cwd()

METADATA_DIR = PROJECT_DIR / "metadata"
OUT_DIR = PROJECT_DIR / "results" / "05_alphafold3_analysis"
INPUT_DIR = OUT_DIR / "input"
AF3_OUTPUT_DIR = INPUT_DIR / "af3_outputs"

AF3_DESIGNS = METADATA_DIR / "af3_designs.tsv"
MOTIF_POSITION_MAP = METADATA_DIR / "af3_motif_position_map.tsv"
BETA2_RESIDUE_MAP = METADATA_DIR / "af3_beta2_residue_map.tsv"

AF3_JSON_DIR = OUT_DIR / "af3_input_json"
ATOM_TABLE_DIR = OUT_DIR / "af3_atom_tables"
PLOTS_DIR = OUT_DIR / "plots"

MANIFEST = OUT_DIR / "af3_output_manifest.tsv"
CONFIDENCE_SUMMARY = OUT_DIR / "af3_confidence_summary.tsv"

# Distance threshold, in Angstrom, used for protein-DNA heavy-atom contact counting.
CONTACT_DISTANCE_A = 4.0

# DNA atom classes used for contact summaries.
# Base atoms are used because the manuscript focuses on contacts to DNA bases at motif positions.
DNA_BACKBONE_ATOMS = {
    "P", "OP1", "OP2", "O1P", "O2P", "O3'", "O5'", "C5'", "C4'", "O4'", "C3'", "C2'", "C1'"
}

# Protein backbone atoms are excluded from beta2 side-chain distance summaries.
PROTEIN_BACKBONE_ATOMS = {"N", "CA", "C", "O", "OXT"}

for path in [OUT_DIR, INPUT_DIR, AF3_OUTPUT_DIR, AF3_JSON_DIR, ATOM_TABLE_DIR, PLOTS_DIR]:
    path.mkdir(parents=True, exist_ok=True)
